### PR TITLE
Detect csharp kernels when opening a notebook

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4061

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/test/datascience/notebook/nonPythonKernels.vscode.test.ts [hunk 1]
```diff
@@ -101,6 +111,10 @@ suite('DataScience - VSCode Notebook - Kernels (non-python-kernel) (slow)', () =
         await openNotebook(api.serviceContainer, testJuliaNb.fsPath);
         await waitForKernelToGetAutoSelected('julia');
     });
+    test('Automatically pick csharp kernel when opening a Julia Notebook', async () => {
+        await openNotebook(api.serviceContainer, testJuliaNb.fsPath);
```
Comment (joyceerhl) [inline src/test/datascience/notebook/nonPythonKernels.vscode.test.ts:115]: I'm confused, why would we want to automatically pick a C# kernel when opening a Julia notebook?

Comment (DonJayamanne) [reply src/test/datascience/notebook/nonPythonKernels.vscode.test.ts:115]: Fixed

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -562,9 +563,22 @@ jobs:
...
+          dotnet-version: ${{env.DOTNET_VERSION}}
+        if: matrix.test-suite != 'notebookWithoutPythonExt'
+
+      - name: Install .NET Interactive
+        run: dotnet tool install -g --add-source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" Microsoft.dotnet-interactive
+        if: matrix.test-suite != 'notebookWithoutPythonExt'
+
+      - name: Install .NET Kernel
+        run: dotnet interactive jupyter install
+        if: matrix.test-suite != 'notebookWithoutPythonExt'
```
Comment (joyceerhl) [inline .github/workflows/pr.yml:580]: Should this be `if: matrix.test-suite == notebookWithoutPythonExt`? That test suite is for testing our notebooks without the Python extension, right?

File: src/test/datascience/notebook/nonPythonKernels.vscode.test.ts [hunk 3]
```diff
@@ -101,6 +111,18 @@ suite('DataScience - VSCode Notebook - Kernels (non-python-kernel) (slow)', () =
         await openNotebook(api.serviceContainer, testJuliaNb.fsPath);
         await waitForKernelToGetAutoSelected('julia');
     });
+    test('Automatically pick csharp kernel when opening a Julia Notebook', async function () {
```
Comment (joyceerhl) [inline src/test/datascience/notebook/nonPythonKernels.vscode.test.ts:114]:
```suggestion
    test('Automatically pick csharp kernel when opening a csharp notebook', async function () {
```

File: src/test/datascience/notebook/nonPythonKernels.vscode.test.ts [hunk 4]
```diff
@@ -151,6 +173,21 @@ suite('DataScience - VSCode Notebook - Kernels (non-python-kernel) (slow)', () =
 
         assertHasTextOutputInVSCode(cell, '123456', 0, false);
     });
+    test('Can run a CSharp notebook', async function () {
+        // Disabled, as .NET interactive kernel does not run on CI.
+        // https://github.com/microsoft/vscode-jupyter/issues/4061#issuecomment-740188708
+        return this.skip();
+        this.timeout(30_000); // Can be slow to start Julia kernel on CI.
```
Comment (joyceerhl) [inline src/test/datascience/notebook/nonPythonKernels.vscode.test.ts:180]: Is the timeout necessary here? If it is,
```suggestion
        this.timeout(30_000); // Can be slow to start csharp kernel on CI.
```

File: .github/workflows/pr.yml (hunk 2 above)
Comment (DonJayamanne) [reply .github/workflows/pr.yml:580]: Nope this is correct. .NET kernel cannot be installed if you do not have Jupyter.

**Review by DavidKutu — APPROVED** [review]

**Pull request merged**